src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/SQLClause.java

```diff
@@ -36,6 +38,7 @@
 abstract class SQLClause<T> {
 
     protected final T expr;
+    final int nestedPathIndex = 1;
```

Comment:
I was saying put this in the method as final var. It's too specific to be on the class level, however it's still ok, but then you must make it static.

Reply:
Done.